src/journal/journald-kmsg.c

```diff
@@ -292,7 +292,8 @@ static void dev_kmsg_record(Server *s, const char *p, size_t l) {
 
                 /* Avoid any messages we generated ourselves via
                  * log_info() and friends. */
-                if (pid && is_us(pid))
+                if (pid && is_us(pid) &&
+                    streq(identifier, program_invocation_short_name))
```

Comment:
`identifier` might be null here. That case needs to be handled the same as before, I think.

Reply:
hmm, let's please move this into the is_us() function, i.e. maybe rework that function to take both pid and identifier as arguments, and let it figure out the details. The check of the process name is a bit ugly I think and thus it would be nicer if it would be hidden within is_us()...

Also, the check for pid being non-NULL should move into is_us() too